tensorbay/client/gas.py

```diff
@@ -616,10 +616,9 @@ def upload_dataset(
         else:
             target_branch_name = branch_name if branch_name else dataset_client.status.branch_name
 
-            for draft in dataset_client.list_drafts():
-                if draft.branch_name == target_branch_name:
-                    dataset_client.checkout(draft_number=draft.number)
-                    break
+            drafts = dataset_client.list_drafts(branch_name=target_branch_name)
```

Comment:
is that possible that `list_drafts` returns multiple drafts?

Reply:
Here we set status = "OPEN", branch_name=target_branch_name, list_drafts will only return one draft at most.